xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-tree/xwiki-platform-index-tree-api/src/main/resources/org/xwiki/index/tree/internal/nestedpages/query/queries.hbm.xml

```diff
@@ -72,7 +72,7 @@
             left outer join xwikidoc doc on (
                 doc.XWD_WEB = XWS_REFERENCE and
                 doc.XWD_NAME = 'WebHome' and
-                doc.XWD_LANGUAGE = '')
+                (doc.XWD_LANGUAGE = '' OR doc.XWD_LANGUAGE is NULL))
```

Comment:
We need a comment to explain these changes  so that we remember them. Thanks!

Reply:
It should also explain why it's not a problem with other DBs.. hmm I wonder if it is. I see in the HBM file that XWD_LANGUAGE could be null:

`<property name="language" type="string" column="XWD_LANGUAGE" index="DOC_LANGUAGE" length="5" />`  

?

Reply:
It technically can be null but it never is in practice because XWikiDocument#getLanguage() never return null.

Reply:
@ClemensRobbenhaar can you add the comment that @vmassol requested. Unfortunately you can't add it in-line but only before the full query. Still useful though.

Reply:
Sorry for the delay. Comments have been added.

Reply:
> unless a not-null constraint is present

Really ? I tough it Oracle was always behaving like this. So that would be that a cleaner fix would be to add this constraint then ? After all this is is never supposed to be null.

Reply:
It sure would be a cleaner fix to add the constraint to the hibernate definition of the schema in ``.../xwiki-platform-oldcore/src/main/resources/xwiki.oracle.hbm.xml`` but I have no idea how to update the existing Oracle databases where currently a ``null`` is stored for the ``XWD_LANGUAGE`` field instead of an empty string.

Reply:
Not sure about the exact request but it would be done in an hibernate migration.

Reply:
I thought the hibernate migrations only run after the schema has been validated/updated which would be too late here, but I see some liquibase migrations can be run before that.
If I write the migration this might need some time as it is my first DB migration for XWiki and also testing it with Oracle might be ... time consuming.

Reply:
Why too late ? The automated hibernate migrator is very (very) basic: all it does is adding missing columns. So what will happen is that the automatic migrator will run (and do nothing) and then in your migrator you will alter the column to make it not nullable.

Reply:
Ah, I thought hibernate checks the schema at startup and fails if not all constraints are present. You see, I have only a very vague idea how the database migrations work in XWiki.